src/main/xsl/preprocess/maprefImpl.xsl

```diff
@@ -389,6 +394,25 @@ See the accompanying LICENSE file for applicable license.
     </xsl:call-template>
   </xsl:template>
 
+  <xsl:template match="@*" mode="preserve-submap-attributes">
+    <xsl:attribute name="dita-ot:submap-{local-name()}" select="."/>
+  </xsl:template>
+  
+  <xsl:template match="*[contains(@class,' topic/title ')]" mode="preserve-submap-title-and-topicmeta">
+    <submap-topicmeta class="+ map/topicmeta ditaot-d/submap-topicmeta ">
+      <submap-title class="+ topic/navtitle ditaot-d/submap-title ">
+        <xsl:apply-templates select="@*" mode="preserve-submap-attributes"/>
+        <xsl:apply-templates select="*|processing-instruction()|text()"/>
+      </submap-title>
+    </submap-topicmeta>
+  </xsl:template>
+  <xsl:template match="*[contains(@class,' map/topicmeta ')]" mode="preserve-submap-title-and-topicmeta">
```

Comment:
We still have the `<foreign>` element here. Not the output is

```xml
<submap>
  <submap-topicmeta>
    <submap-title>...</submap-title>
  </submap-topicmeta>
  <submap-topicmeta-container>
    ...
  </submap-topicmeta-container>
  ...
</submap>
```

What's the reason for not doing simply

```xml
<submap>
  <submap-topicmeta>
    <submap-title>...</submap-title>
    <submap-topicmeta-container>
      ...
    </submap-topicmeta-container>
  </submap-topicmeta>
  ...
</submap>
```

And if we do that, we don't really need `<submap-topicmeta-container>` anymore?